local/src/test/scala/com/salesforce/op/local/OpWorkflowModelLocalTest.scala

```diff
@@ -169,6 +165,36 @@ class OpWorkflowModelLocalTest extends FlatSpec with PassengerSparkFixtureTest w
     }
   }
 
+  private def assertLoadModelAndScore(
+    modelLocation: String,
+    rawData: Array[Map[String, Any]],
+    expectedScores: Array[(Prediction, RealNN, RealNN, Text)],
+    prediction: FeatureLike[Prediction]
+  ): Unit = {
+    val scoreFn = OpWorkflowModel.load(modelLocation).scoreFunction
+    scoreFn shouldBe a[ScoreFunction]
+    val scores = rawData.map(scoreFn)
+    assert(scores, expectedScores, prediction)
+  }
+
+  private def buildAndSaveModel(modelsAndParams: Seq[(EstimatorType, Array[ParamMap])]) = {
+    val prediction = BinaryClassificationModelSelector.withTrainValidationSplit(
+    modelsAndParameters = modelsAndParams, splitter = None
+    ).setInput(survivedNum, features).getOutput()
+    val workflow = new OpWorkflow().setReader(dataReader)
+    .setResultFeatures(prediction, survivedNum, indexed, deindexed)
+    lazy val model = workflow.train()
+    val path = Paths.get(tempDir.toString, "op-runner-local-test-model").toFile.getCanonicalFile.toString
+    model.save(path)
+    (path, model, prediction)
+  }
+
+  private def genRawDataAndScore(model: OpWorkflowModel, prediction: FeatureLike[Prediction]) = {
+    lazy val rawData = dataReader.generateDataFrame(model.getRawFeatures()).sort(KeyFieldName).collect().map(_.toMap)
```

Comment:
I don't actually know, but do these local variables need to be lazy if the method itself is assigned to a lazy val?

Reply:
good point, let me try removing lazy val inside function def :)